File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        q=None
        c=0
        while l1!=None or l2!=None:
            if l1 is None:
                p=0
            else:
                p=l1.val
                l1=l1.next
            if l2 is None:
                r=0
            else:
                r=l2.val
                l2=l2.next
            s=p+r+c
            c=s//10 if s>9 else 0
            s%=10
            
            d=ListNode(s)
            if q is None:
                q=d
                g=d
            else:
                g.next=d
                g=g.next
        
        if c>0:
            g.next=ListNode(c)
            g=g.next
        return q
            
            

File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode end= new ListNode(0,null);
        ListNode temp1=l1,temp2=l2,head = end;
        int carry=0;
        while(temp1!=null || temp2!=null)
        {
            int val=0;
            if(temp1 != null){ 
                val += temp1.val;
                temp1 = temp1.next;
            }
            if(temp2 != null) {
                val += temp2.val;
                temp2 = temp2.next;
            }
            int temp = val + carry;
            val = temp % 10;
            carry = temp / 10;         
            ListNode l=new ListNode(val,null);
            end.next=l;
            end=l;
            
        }
        if(carry==1)
        {
            ListNode l=new ListNode(1,null);
            end.next=l;
            end=l;
        }
        return head.next;
    }
}